### ISSUE TITLE
Programmatically set the App Store "Country/Region" 

### ISSUE THREAD

**Issue by elliotweiser** [mas-cli/mas#75]
# Summary

I've been doing some Ansible role/playbook development, with the following goal: automate the complete provisioning/rebuild of my workstation. Since I work primarily on a Mac (OS X v10.12.2), the `mas` CLI is extremely useful for automatically installing apps distributed through the App Store.

So far, the project is going well. I've been using the [jhcook OS X Sierra box](https://atlas.hashicorp.com/jhcook/boxes/macos-sierra/versions/10.12.2) for testing with vagrant/virtualbox. However I've encountered a couple issues with using `mas` when testing.

Invoking the `mas signin <USERNAME> <PASSWORD>` command, a dialog pops up in the App Store GUI, requiring an acknowledgement of `OK` that the App Store should go ahead and update the Country/Region according to the signed-in user's Apple account's billing settings. Presumably, this is because the Country/Region is either unset or doesn't match my Apple account's billing settings. The `mas` CLI is unusable until that dialog has been completed. Below is the complete dialog text:

```
This Apple ID is only valid for purchases in the U.S. iTunes Store. You will be switched to that Store.
```

#### How to reproduce

I believe this can be reproduced with the following steps:

1. Open the App Store
2. Navigate to the Purchases tab
3. Select the flag icon in the bottom right corner
4. Select any Country/Region other than that which is associated with your account.
5. Run `mas signin ...`

#### Follow-up questions

1. Has anyone else run into this issue?
2. Does anyone know of a good way to programmatically set the App Store's Country/Region?
3. Are there potentially any "bad" ways, such as modifying a plist file that is loaded into the App Store's runtime?

If there are no workarounds, would anyone be willing to implement another CLI command to `mas` e.g. `mas set-country <COUNTRY>`? It could also be added as a `--country` option to the `mas signin` command, as long as setting the country occurs prior to authentication.

Also, since it is my first time reaching out to the developers and maintainers of this project, I'd like to say thank you for reading this issue and working on this project 😄 .

**Comment by rodionovd** [comment]
Hello Elliot! Thank you for this awesome report!

> Has anyone else run into this issue?

Yes, this is indeed an issue that happens when you try to sign-in into an account that is associated with different country/region from what your Mac App Store.app (and probably your Mac) has been set to.

> Does anyone know of a good way to programmatically set the App Store's Country/Region?

I'll definitely look into it because I'd like to implement the `mas set-country` command you were talking about — I think that'd be very helpful!

> Are there potentially any "bad" ways, such as modifying a plist file that is loaded into the App Store's runtime?

Not sure if there's any plist file loaded at all…

> Also, since it is my first time reaching out to the developers and maintainers of this project, I'd like to say thank you for reading this issue and working on this project 😄 .

Oh you're very welcome! 😁

**Comment by elliotweiser** [comment]
Thank you, for responding @rodionovd. I think implementing `mas set-country` command would be an excellent new feature. Much appreciated 👍

**Comment by elliotweiser** [comment]
Hi @rodionovd! Any updates on this?

**Comment by phatblat** [comment]
@elliotweiser this is currently blocked by #129. We will look into it soon, but need to understand the API changes in High Sierra.

**Comment by elliotweiser** [comment]
@phatblat Any chance this can be looked into now that #129 has been addressed?

**Comment by phatblat** [comment]
#129 isn't exactly "fixed", the `signin` command is disabled on 10.13+ to prevent the crash. #164 was opened to continue the conversation since there were several related issues opened in the past.

This is still an important feature as the tool should be aware of user and app countries. I'll keep an eye out for APIs that would give it visibility and/or control over the country. Ideally, there should be no GUI involved but sometimes they get triggered automatically by Apple's MAS logic.

**Comment by elliotweiser** [comment]
Ah okay I'll follow that issue. Thanks for the update!

**Comment by chris-araman** [comment]
My understanding is that the App Store region is tied to an Apple ID. If it is possible for `mas` to change the region programmatically, it would be changed on the Apple ID for all services, not just the App Store. There are implications for a user's existing purchases, memberships, subscriptions, balances, and downloaded books, music, movies, and TV shows.
https://support.apple.com/en-us/HT201389

We would want to be very careful in how this is communicated to the user before implementing this feature request.

**Comment by rgoldberg** [comment]
Can this be closed as a duplicate of #444, which is more up-to-date?
